sdk/storage/inc/common/storage_error.hpp

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <map>
+#include <stdexcept>
+#include <string>
+
+namespace Azure { namespace Core { namespace Http {
+  class Response;
+}}} // namespace Azure::Core::Http
+
+namespace Azure { namespace Storage {
+
+  struct StorageError : public std::runtime_error
```

Comment:
We should discuss about design of this StorageError

Reply:
Discussed and found a happy design for both of us :).